spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/SubscriberFactory.java

```diff
@@ -28,9 +28,17 @@
  * @author João André Martins
  * @author Mike Eltsufin
  * @author Artem Bilan
+ * @author Doug Hoard
  */
 public interface SubscriberFactory {
 
+	/**
+	 * Method to get the project id
+	 * @return the project id
+	 * @since 1.1
+	 */
+	String getProjectId();
```

Comment:
So, what is the reason to have all these new overhead with the `projectId`?

Where is the reason in the commit message?
I didn't get its requirements from the code, too.

Please, elaborate

Thanks

Reply:
The method `getProjectId()` was added so that the `PubSubSubscriberTemplate` can add the `projectId` to messages.

This allows the `PubSubSubscriberTemplate` batch ack 
 method (`ack(Collection<AcknowledgeablePubsubMessage> acknowledgeablePubsubMessages)`) to be able to validate the messages can be acked by the `PubSubSubscriberTemplate`.